ext/hook/uhook.c

```diff
@@ -250,6 +258,37 @@ static bool dd_uhook_begin(zend_ulong invocation, zend_execute_data *execute_dat
     }
     dyn->hook_data->execute_data = NULL;
 
+    if (dyn->hook_data->suppress_call) {
+        if (ZEND_USER_CODE(execute_data->func->type)) {
+            static union {
+                zend_op zop;
+                zval zv;
+            } retop[] = {[0].zop =
+                             {
+                                 .opcode = ZEND_RETURN,
+                                 .op1_type = IS_CONST,
+                                 .op1 = {.constant = sizeof(retop[0])},
+                                 .op2_type = IS_UNUSED,
+                             },
+                         [1].zv = {.u1.type_info = IS_NULL}};
+            // the race condition doesn't matter
```

Comment:
which race condition? Comment seems misleading

Reply:
If we define it as a situation where two threads access shared data without synchronization and at least one is a write, then this code is clearly racy in ZTS (`zend_vm_set_opcode_handler(&retop[0].zop)` can be called twice, but even one thread writing the opcode handler and the other reading a non-NULL value would be a race condition). However, the only two possible values read for the opcode handler are a NULL value and the correct non-NULL value (at least on amd64/arm64, where aligned 64-bit writes cannot be partially seen).

Reply:
On that note, is there a reason for this to not just simply be a true global initialized in zai_uhook_minit?

Reply:
Is that enough though? Can't it be set in the startup hook of another zend extension?

Reply:
Let's leave it as is, it's good enough here. Let's not make it more complicated than it needs to be.